### ISSUE TITLE
Error uploading gif image in Story Builder

### ISSUE THREAD

**Issue by jwirfs-brock** [denverfoundation/storybase#832]
When I tried to upload the attached gif images, I received an error message:

{"error_message": "Sorry, this request could not be processed. Please try again later."}

Here are the images that caused the error:

![fl_login9](https://f.cloud.github.com/assets/1020518/932206/21cfd00e-003c-11e3-9c6e-538e4c4d5c73.gif)
![fl_login13](https://f.cloud.github.com/assets/1020518/932208/252ac556-003c-11e3-8d32-4c20c9a5f356.gif)

**Comment by ghing** [comment]
I believe the traceback for this looks something like this:

```
Traceback (most recent call last):

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 217, in wrapper
    response = callback(request, *args, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 459, in dispatch_list
    return self.dispatch('list', request, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 491, in dispatch
    response = method(request, **kwargs)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 1363, in post_list
    updated_bundle = self.full_dehydrate(updated_bundle)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/tastypie/resources.py", line 871, in full_dehydrate
    bundle.data[field_name] = method(bundle)

  File "/home/floodlight/webapps/floodlight_production/floodlight/apps/storybase_asset/api.py", line 274, in dehydrate_thumbnail_url
    return bundle.obj.get_thumbnail_url(width=222, height=222)

  File "/home/floodlight/webapps/floodlight_production/floodlight/apps/storybase_asset/models.py", line 579, in get_thumbnail_url
    thumbnail = thumbnailer.get_thumbnail(thumbnail_options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/files.py", line 396, in get_thumbnail
    thumbnail = self.generate_thumbnail(thumbnail_options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/files.py", line 302, in generate_thumbnail
    quality=quality).read()

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/easy_thumbnails/engine.py", line 55, in save_image
    image.save(destination, format=format, **options)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/PIL/Image.py", line 1439, in save
    save_handler(self, fp, filename)

  File "/home/floodlight/webapps/floodlight_production/venv/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 425, in _save
    raise IOError("cannot write mode %s as JPEG" % im.mode)

IOError: cannot write mode P as JPEG
```

**Comment by ghing** [comment]
This report sounds really similar to #726.  I don't understand why the fix for that doesn't work here.

**Comment by jwirfs-brock** [comment]
When I re-saved the image file as a PNG, I was able to upload it successfully.

**Comment by ghing** [comment]
The images @jwirfs-brock provided that caused this issue were using the indexed color mode.  I tried converting to RGB and confirmed that the problem persisted.

I tried to replicate the issue on my notebook and the image upload (using the original image) worked fine.  I thought it might be because my laptop is using Pillow (1.7.8) while dev is using PIL (1.1.7), but production is using Pillow (1.7.7), so I don't think that's the problem.

Perhaps Pillow/PIL, or some underlying library doesn't have GIF support in dev/production.  I'm going to try to figure out how to check for this support.

**Comment by ghing** [comment]
I found a discrepancy between the different environments. The local version has django-filer==0.9.4a1.dev1, dev has django-filer==0.9.3, production has django-filer==0.9.2a1.dev1. Upgrading the version on dev to django-filer==0.9.4 seemed to fix the issue.

**Comment by ghing** [comment]
~~I tried upgrading django-filer to 0.9.4 on staging and the issue persisted.~~

**Update:** I thought I had upgraded, django-filer to 0.9.4, but this wasn't the case because of the existence of the django-filer 0.9.2 build directory and the behavior [described here](http://stackoverflow.com/questions/14617136/why-is-pip-installing-an-old-version-of-my-package).

I had to remove the build directory and then install the desired version of django-filer.  After that, gif uploads worked!

```
rm -rf ../venv/build/django-filer/
pip install django-filer==0.9.5
```
